src/Application/Console/Formatters/Checkstyle.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Formatters;
+
+use Exception;
+use NunoMaduro\PhpInsights\Application\Console\Contracts\Formatter;
+use NunoMaduro\PhpInsights\Domain\Contracts\HasDetails;
+use NunoMaduro\PhpInsights\Domain\Details;
+use NunoMaduro\PhpInsights\Domain\Insights\Insight;
+use NunoMaduro\PhpInsights\Domain\Insights\InsightCollection;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class Checkstyle implements Formatter
+{
+    /** @var OutputInterface */
+    private $output;
+
+    public function __construct(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+    }
+
+    /**
+     * Format the result to the desired format.
+     *
+     * @param InsightCollection $insightCollection
+     * @param string $dir
+     * @param array<string> $metrics
+     */
+    public function format(
+        InsightCollection $insightCollection,
+        string $dir,
+        array $metrics
+    ): void
+    {
+        $checkstyle = new \SimpleXMLElement('<checkstyle/>');
```

Comment:
Should we also make a check here to see if `SimpleXMLElement` is available and if not, then throw an exception with a user friendly error message as to why it failed?